### PR TITLE
First miner feature tests

### PR DESCRIPTION
**Pull request by Neurosploit**
-Add test project for miner feature
-Tests for MinerController
-Tests for MiningRPCController
-Attempt to fix locally failing LightWalletManager tests

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.Miner/Controllers/MinerController.cs [hunk 1]
```diff
@@ -118,5 +115,17 @@ public IActionResult StartStaking([FromBody]StartStakingRequest request)
                 return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, e.Message, e.ToString());
             }
         }
+
+        private Wallet.Wallet GetWallet(IWalletManager walletManager, StartStakingRequest request)
```
Comment (Aprogiena) [inline src/Stratis.Bitcoin.Features.Miner/Controllers/MinerController.cs]: please no new code outside tests without xmldoc

File: src/Stratis.Bitcoin.Features.Miner/MiningRPCController.cs [hunk 2]
```diff
@@ -149,5 +141,40 @@ public GetStakingInfoModel GetStakingInfo(bool isJsonFormat = true)
...
+            if (account == null)
+                throw new RPCServerException(NBitcoin.RPC.RPCErrorCode.RPC_INVALID_REQUEST, "No account found on wallet");
+
+            var res = new WalletAccountReference(walletName, account.Name);
+
+            this.logger.LogTrace("(-):'{0}'", res);
+            return res;
+        }
+
+        private Wallet.Wallet GetWallet(IWalletManager walletManager, string walletName)
```
Comment (Aprogiena) [inline src/Stratis.Bitcoin.Features.Miner/MiningRPCController.cs]: xmldoc

File: src/Stratis.Bitcoin.IntegrationTests/APITests.cs [hunk 3]
```diff
@@ -93,7 +94,7 @@ public void CanStartStakingViaAPI()
                 var fullNode = this.apiTestsFixture.stratisStakeNode.FullNode;
                 var ApiURI = fullNode.Settings.ApiUri;
 
-                Assert.NotNull(fullNode.NodeService<PosMinting>(true));
+                    Assert.NotNull(fullNode.NodeService<IPosMinting>(true));
```
Comment (Aprogiena) [inline src/Stratis.Bitcoin.IntegrationTests/APITests.cs]: why the extra spaces?

Comment (Neurosploit) [reply src/Stratis.Bitcoin.IntegrationTests/APITests.cs]: Appeared during rebasing

File: src/Stratis.Bitcoin.Features.Miner/Controllers/MinerController.cs (hunk 1 above)
Comment (dangershony) [reply src/Stratis.Bitcoin.Features.Miner/Controllers/MinerController.cs]: Ah @Neurosploit you've been caught

**Pull request closed without merging**